extensions/HuggingFaceTransformers/python/src/aiconfig_extension_hugging_face_transformers/text_2_image.py

```diff
@@ -0,0 +1,341 @@
+        prompt: Prompt,
+        aiconfig: "AIConfigRuntime",
+        _options,
+        params: Optional[Dict[str, Any]] = {},
+    ) -> Dict[str, Any]:
+        """
+        Defines how to parse a prompt in the .aiconfig for a particular model
+        and constructs the completion params for that model.
+
+        Args:
+            serialized_data (str): Serialized data from the .aiconfig.
+
+        Returns:
+            dict: Model-specific completion parameters.
+        """
+        # Build Completion data
+        model_settings = self.get_model_settings(prompt, aiconfig)
+        [_pipeline_creation_params, unfiltered_completion_params] = refine_pipeline_creation_params(model_settings)
+        completion_data = refine_image_completion_params(unfiltered_completion_params)
+        
+        #Add resolved prompt
+        resolved_prompt = resolve_prompt(prompt, params, aiconfig)
+        completion_data["prompt"] = resolved_prompt
+        return completion_data
+
+    async def run_inference(
+        self, prompt: Prompt, aiconfig : "AIConfigRuntime", options : InferenceOptions, parameters: Dict[str, Any]
+    ) -> List[Output]:
+        """
+        Invoked to run a prompt in the .aiconfig. This method should perform
+        the actual model inference based on the provided prompt and inference settings.
+
+        Args:
+            prompt (str): The input prompt.
+            inference_settings (dict): Model-specific inference settings.
+
+        Returns:
+            InferenceResponse: The response from the model.
+        """
+        model_settings = self.get_model_settings(prompt, aiconfig)
+        [pipeline_creation_data, _] = refine_pipeline_creation_params(model_settings)
+        if not pipeline_creation_data.get("requires_safety_checker", True):
+            pipeline_creation_data["safety_checker"] = None
+
+        pipeline_building_disclaimer_message = """
+Building the pipeline... This can take a long time if you haven't created one before 
+on this machine. Please be patient!
+
+If this seems to be taking too long, you can change some of the pipeline generation
+params. See more details here:
+https://huggingface.co/docs/diffusers/using-diffusers/loading
+"""
+        print(pipeline_building_disclaimer_message)
+
+        model_name : str = aiconfig.get_model_name(prompt)
+        # TODO (rossdanlm): Figure out a way to save model and re-use checkpoint
+        # Otherwise right now a lot of these models are taking 5 mins to load with 50
+        # num_inference_steps (default value). See here for more details:
+        # https://huggingface.co/docs/diffusers/using-diffusers/loading#checkpoint-variants
+        if isinstance(model_name, str) and model_name not in self.generators:
+            self.generators[model_name] = AutoPipelineForText2Image.from_pretrained(
+                pretrained_model_or_path=model_name,
+                safety_checker=None,
+                **pipeline_creation_data
+            )
```

Comment:
not sure how we're handling memory/hardware requirements but it would probably make sense to utilize a gpu if available.

something like
```python
device = torch.device("cuda" if torch.cuda.is_available() else "cpu")

# Move to device.
# pipeline = AutoPipelineForText....
pipeline.to(device)
```

There's also the metal framework on Mac, the pytorch device is called "mps" and it runs a lot faster than on cpu for local testing.

Reply:
Ok let me do that in future diff stack!